packages/eslint-plugin/src/rules/strict-enums.ts

```diff
@@ -0,0 +1,806 @@
+import * as util from '../util';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import { TSESTree } from '@typescript-eslint/utils';
+
+/**
+ * TypeScript only allows number enums, string enums, or mixed enums with both
+ * numbers and strings.
+ *
+ * Mixed enums are be a union of a number enum and a string enum, so there is
+ * no separate kind for them.
+ */
+enum EnumKind {
+  NON_ENUM,
+  HAS_NUMBER_VALUES,
+  HAS_STRING_VALUES,
+}
+
+/** These operators are always considered to be safe. */
+const ALLOWED_ENUM_OPERATORS = new Set(['in', '|', '&', '|=', '&=']);
```

Comment:
nit XOR `^` is another bitwise people often use (to subtract values from a flags enum - i.e. the inverse of OR `|`)
```
1 | 2 === 3 // add values
3 & 2 === 2 // check for values
3 ^ 2 === 1 // remove values
```


```suggestion
const ALLOWED_ENUM_OPERATORS = new Set(['in', '|', '&', '^', '|=', '&=', '^=']);
```